gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -230,7 +264,47 @@ public GrpcCallContext withExtraHeaders(Map<String, List<String>> extraHeaders)
         streamWaitTimeout,
         streamIdleTimeout,
         channelAffinity,
-        newExtraHeaders);
+        newExtraHeaders,
+        retrySettings,
+        retryableCodes);
+  }
+
+  @Override
+  public RetrySettings getRetrySettings() {
+    return this.retrySettings;
+  }
+
+  @Override
```

Comment:
What is the difference between the retry settings here, and those in UnaryCallSettings?

Reply:
* The `RetrySettings` in `UnaryCallSettings` are the default retry settings that are generated as part of the gapic client code generation. These settings are used if the client application does not set any `RetrySettings` on the `CallContext` that is used for an invocation of the RPC.
* The `RetrySettings` that are set on the `CallContext` will override the settings in `UnaryCallSettings` for the invocation(s) where the `CallContext` is used.

Reply:
Furthermore, `UnaryCallSettings` are available at "client-creation", not at run time. As @olavloite mentioned, the `UnaryCallSettings` supply the default settings for a `Callable` at *creation*. The `RetrySettings` are extracted from the `UnaryCallSettings` and used throughout the chain independently.